src/main/kotlin/com/github/biomejs/intellijbiome/services/BiomeServerService.kt

```diff
@@ -124,18 +125,39 @@ class BiomeServerService(private val project: Project) {
 
             val formattingResults = server.sendRequest { it.textDocumentService.formatting(formattingParams) }
 
-            formattingResults?.forEach {
-                val lineSeparator = StringUtil.detectSeparators(it.newText)
-                val normalizedText = StringUtil.convertLineSeparators(it.newText)
+            formattingActions.add(Runnable {
+                var lineSeparator: LineSeparator? = null
+
+                formattingResults?.forEach {
```

Comment:
I'm just wondering if we need to sort text edits.
It seems that we mutate a document text and if we start with the top, we can get incorrect offsets.
What do you think if we need to apply them in a reverse order?

Reply:
I missed that part in the referenced code: https://github.com/ballerina-platform/lsp4intellij/blob/3bd015cfd2772ecdc06675704024c7d30e242a4a/src/main/java/org/wso2/lsp4intellij/editor/EditorEventManager.java#L1257-L1258. Indeed we need to do in a reversed order. I'll fix it in another PR.